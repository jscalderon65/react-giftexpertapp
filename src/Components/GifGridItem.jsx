import React from "react";
import PropTypes from "prop-types";
const GifGridItem = ({img}) => {
  const { url, title } = img;
  return (
      <div  style={{ width:"200px",
      height:"auto"}}>
          <img 
            width="200px"
            height="200px"
            src={url}
            alt={title}
            />
            <hr/>
            <p>{title}</p>
      </div>
  );
};
export default GifGridItem;
GifGridItem.propTypes ={
  img: PropTypes.object.isRequired,
}
