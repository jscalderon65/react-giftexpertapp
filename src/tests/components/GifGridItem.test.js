import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../Components/GifGridItem";

describe("Pruebas en <GifGridItem />", () => {
  const img = {
    url: "https://sitechecker.pro/wp-content/uploads/2017/12/URL-meaning.png",
    title: "sisas",
  };
  const wrapper = shallow(<GifGridItem img={img} />);
  test("Se debe mostrar correctamente el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe tener un pÃ¡rrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(img.title);
  });
  test("Debe tener la url del objeto img", () => {
    const image = wrapper.find("img");
    expect(image.props().src.trim()).toBe(img.url);
    expect(image.props().alt.trim()).toBe(img.title);
  });
});
