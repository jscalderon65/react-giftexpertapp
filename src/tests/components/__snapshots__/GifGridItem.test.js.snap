// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Se debe mostrar correctamente el componente 1`] = `
<div
  style={
    Object {
      "height": "auto",
      "width": "200px",
    }
  }
>
  <img
    alt="sisas"
    height="200px"
    src="https://sitechecker.pro/wp-content/uploads/2017/12/URL-meaning.png"
    width="200px"
  />
  <hr />
  <p>
    sisas
  </p>
</div>
`;
